broker.yaml

```diff
@@ -0,0 +1,728 @@
+                  type: string
+                  description: 'URL of service registry which when requested returns an array of all services under organisation each object must containing following schema : {project_id, callback_url, access_token}'     
+      responses:
+        '200':
+          description: "List of subscription id's"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                      type: string
+                      description: subscriptionId
+                      example: subscriptionId
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+        default:
+          $ref: '#/components/responses/Error'
+    delete:
+      summary: 'Delete service registry'
+      operationId: d
+      tags:
+        - services-from-registry
+      parameters:
+        - in: cookie
+          name: userAccessToken
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: 'The service registry is removed successfully'
+          content:
+            application/json:
+              schema:
+                type: string
+                example: "Deleted successfully"
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+        default:
+          $ref: '#/components/responses/Error'  
+components:
+  securitySchemes:
+    bearerAuth:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+  responses:
+    Unauthorized:
+      description: 'Unauthorized ([RFC 7235](https://tools.ietf.org/html/rfc7235))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    Forbidden:
+      description: 'Forbidden ([RFC 7231](https://tools.ietf.org/html/rfc7231))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    NotFound:
+      description: 'Not found ([RFC 7231](https://tools.ietf.org/html/rfc7231))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    InternalServerError:
+      description: 'Internal server error ([RFC 7231](https://tools.ietf.org/html/rfc7231))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+    Error:
+      description: 'Unexpected error ([RFC 7231](https://tools.ietf.org/html/rfc7231))'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+  
+  schemas:
+    Service:
+      type: object
+      properties:
```

Comment:
Just to add on: it may be even better to store all commits up to and including the indicated trusted head commit (and for all updates any future commits up to and including the new head commits), so that, e.g., renaming branches would not automatically invalidate a subscription, as the consistency of a branch would then depend on its constituing member commits and not the branch name.

Reply:
What if history is rewritten? For example a password is committed to master/main.

Reply:
Another corner case is when part of the code is developed by a trusted developer that then leaves. Older commits should be accepted, but not newer ones. Then maybe we should store all commits.